vowpalwabbit/baseline.cc

```diff
@@ -15,47 +17,25 @@ using namespace VW::config;
 namespace
 {
 const float max_multiplier = 1000.f;
-const size_t baseline_enabled_idx = 1357;  // feature index for enabling baseline
 }  // namespace
 
 namespace BASELINE
 {
 void set_baseline_enabled(example* ec)
 {
-  auto& fs = ec->feature_space[message_namespace];
-  for (auto& f : fs)
-  {
-    if (f.index() == baseline_enabled_idx)
-    {
-      f.value() = 1;
-      return;
-    }
-  }
-  // if not found, push new feature
-  fs.push_back(1, baseline_enabled_idx);
+  if (!baseline_enabled(ec)) { ec->indices.push_back(baseline_enabled_message_namespace); }
 }
 
 void reset_baseline_disabled(example* ec)
 {
-  auto& fs = ec->feature_space[message_namespace];
-  for (auto& f : fs)
-  {
-    if (f.index() == baseline_enabled_idx)
-    {
-      f.value() = 0;
-      return;
-    }
-  }
+  auto it = std::find(ec->indices.begin(), ec->indices.end(), baseline_enabled_message_namespace);
+  if (it != ec->indices.end()) { ec->indices.erase(it); }
 }
 
 bool baseline_enabled(example* ec)
 {
-  auto& fs = ec->feature_space[message_namespace];
-  for (auto& f : fs)
-  {
-    if (f.index() == baseline_enabled_idx) return f.value() == 1;
-  }
-  return false;
+  auto it = std::find(ec->indices.begin(), ec->indices.end(), baseline_enabled_message_namespace);
+  return it != ec->indices.end();
```

Comment:
line 190 of the right side (in case you are on diff mode), implies that this reduction only works with the global interactions and not the per example ones, is this an issue?

Reply:
I'm not sure how this reduction is used. It may be an issue.